Automattic/newspack-plugin
# Campaigns Wizard: referrer segmentation

### All Submissions:






### Changes proposed in this Pull Request:



UI for adding a referrer-based segmentation. 

Do not merge before https://github.com/Automattic/newspack-popups/pull/323)%7C is merged.

### How to test the changes in this Pull Request:

1. Switch Campaign plugin to `add/handle-referrer-segment` branch (https://github.com/Automattic/newspack-popups/pull/323)
1. Create a new segment, observe a new section – "Referrer"
2. Set the referrer to `google.com`
3. Edit or create a campaign, assign the new segment to it
4. Verify the campaign is not displayed when site is visited in a fresh session
5. Spoof the referrer by editing [this line](https://github.com/Automattic/newspack-popups/blob/0493d8db6e3609d95a0409359e1515b5f46259ad/includes/class-newspack-popups-inserter.php#L425) in `class-newspack-popups-inserter.php`
6. Observe that the campaign is displayed now
7. Create a segment with multiple referrers, repeat the testing steps to verify the condition works for all of them
8. Test with a referrer-less segment (just to see if a lacking referrer does not break things)

### Other information:
